community/modules/compute/schedmd-slurm-gcp-v5-node-group/main.tf

```diff
@@ -23,10 +23,11 @@ locals {
 
   # Handle VM image format from 2 sources, prioritize source_image* variables
   # over instance_image
-  source_image_input_used = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
-  source_image            = local.source_image_input_used ? var.source_image : lookup(var.instance_image, "name", "")
-  source_image_family     = local.source_image_input_used ? var.source_image_family : lookup(var.instance_image, "family", "")
-  source_image_project    = local.source_image_input_used ? var.source_image_project : lookup(var.instance_image, "project", "")
+  source_image_input_used         = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
+  source_image                    = local.source_image_input_used ? var.source_image : var.instance_image.name
+  source_image_family             = local.source_image_input_used ? var.source_image_family : var.instance_image.family
+  source_image_project            = local.source_image_input_used ? var.source_image_project : var.instance_image.project
+  source_image_project_normalized = strcontains(local.source_image_project, "/") ? local.source_image_project : "projects/${local.source_image_project}/global/images/family"
```

Comment:
I suspect that this conversion is only needed when we are selecting the image based on family (not name). To go further, I think you may see an error when providing this format with image name. If this is the case then we would only want to perform this conversion if 1) there are no slashes in the var and 2) family != "".

Reply:
Added the condition `local.source_image != ""`, so that if the image name is used, the project won't be modified.